airflow/providers/amazon/aws/operators/base_aws.py

```diff
@@ -0,0 +1,97 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import Sequence
+
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.utils.mixins import AwsBaseHookMixin, AwsHookParams, AwsHookType
+
+
+class AwsBaseOperator(BaseOperator, AwsBaseHookMixin[AwsHookType]):
+    """
+    Base AWS (Amazon) Operator Class to build operators on top of AWS Hooks.
+
+    .. warning::
+        Only for internal usage, this class might be changed, renamed or removed in the future
+        without any further notice.
```

Comment:
Regarding this, should we change the class name to `_AwsBaseOperator`?

Same for the Sensor and mixin classes

Reply:
There is kind of tradeoff between common Python's agreement about internal variables/classes/methods and airflow public models. `_` shows that is internal stuff however IDE doesn't like this imports even with the same package (I guess static checks is fine) in the other hand we have some methods and class in core which do no prefixed by `_` but have a description and `:meta private:`.

There is also maybe a good idea to make this stuff public after a while for create custom operators, but for now I would rather keep it internal only

In the end no one could stop end users to use this stuff directly